src/MonoTorrent/MonoTorrent.Client/Modes/DownloadMode.cs

```diff
@@ -65,14 +66,25 @@ public override void Tick (int counter)
 
             UpdateSeedingDownloadingState ();
 
-            for (int i = 0; i < Manager.Peers.ConnectedPeers.Count; i++) {
-                if (!ShouldConnect (Manager.Peers.ConnectedPeers[i])) {
-                    ConnectionManager.CleanupSocket (Manager, Manager.Peers.ConnectedPeers[i]);
-                    i--;
+            PeerId[] arrPID = Manager.Peers.ConnectedPeers.ToArray();
+
+            for (int i = 0; i < arrPID.Length; i++)
+            {
+                try
+                {
+                    if (!ShouldConnect(arrPID[i]))
+                    {
+                        ConnectionManager.CleanupSocket(Manager, arrPID[i]);
+                    }
+                }
+                catch (System.Exception ex)
+                {
+                    Debug.WriteLine("Tick Error " + ex.Message);
                 }
             }
         }
 
```

Comment:
I think the old and new code are functionally the same, except CleanupSocket has been wrapped in a try/catch. Were you seeing exceptions being raised in this method? the expectation for MonoTorrent is that 'CleanupSocket' never raises an error.